zerver/views/streams.py

```diff
@@ -274,6 +274,79 @@ def update_stream_backend(
     # description even for private streams.
     (stream, sub) = access_stream_for_delete_or_update(user_profile, stream_id)
 
+    # Validate that the proposed state for permissions settings is permitted.
+    if is_private is not None:
+        proposed_is_private = is_private
+    else:
+        proposed_is_private = stream.invite_only
+
+    if is_web_public is not None:
+        proposed_is_web_public = is_web_public
+    else:
+        proposed_is_web_public = stream.is_web_public
+
+    if stream.realm.is_zephyr_mirror_realm:
+        # In the Zephyr mirroring model, history is unconditionally
+        # not public to subscribers, even for public streams.
+        proposed_history_public_to_subscribers = False
+    elif history_public_to_subscribers is not None:
+        proposed_history_public_to_subscribers = history_public_to_subscribers
+    elif is_private is not None:
+        # By default, private streams have protected history while for
+        # public streams history is public by default.
+        proposed_history_public_to_subscribers = not is_private
```

Comment:
Added this block to set the correct `history_public_to_subscribers` value for default cases of private and public streams when only `is_private` is passed.

Reply:
That seems correct for backwards-compatibility.